src/sensor/pingsensor.cpp

```diff
@@ -14,6 +14,21 @@ PingSensor::PingSensor()
     connect(dynamic_cast<PingParserExt*>(_parser), &PingParserExt::parseError, this, &PingSensor::parserErrorsUpdate);
 }
 
+void PingSensor::connectLink(const LinkConfiguration conConf, const LinkConfiguration& logConf)
+{
+    Sensor::connectLink(conConf, logConf);
+    if (_parser) {
+        connect(link(), &AbstractLink::newData, this, [this](QByteArray data) {
+            for (char b : data) {
+                Parser::ParserState result = _parser->parseByte(b);
+                if (result == Parser::NEW_MESSAGE) {
+                    handleMessage(_parser->rxMessage());
+                }
+            }
+        });
+        emit this->connectionUpdate();
```

Comment:
`this` is not necessary.

Reply:
Also, how this update the connection ? This is done by the base sensor class, and should be done only when the state of the connection changes, and you are not doing it here.

Reply:
I think I was trying to maintain some order of calls that was already existing. Probably you are right that this does not belong.